Update the contributing page about the GEP process (#2393)
kubernetes-sigs/gateway-api: site-src/contributing/enhancement-requests.md

If you're unsure the proposed work requires an enhancement, file an issue
and ask.

## New Enhancement process

The new enhancement process is documented on the [GEP Overview][gep] page. Please
see that page for all the details about how to log a new GEP, and the process
it will follow on its journey towards Completed status.

> In particular, please note that having a discussion of some form, usually both
> a [Github Discussion][discussion] and some discussion in our [community meetings][meetings]
> is the best way to ensure that your new GEP process will go as smoothly as possible.

Reviewer: I would actually like to be a little stronger with this language.

```suggestion
A **documented** discussion of some form **must** exist prior to submitting a request for enhancement
if that enhancement is non-trivial (which we will define as either: _implicates changes to the API specification_ OR _has some kind of end-user impact_). Please use our [Github Discussions][discussion]
forum as the initial place to start, and feel free to bring that discussion up for synchronous conversation in
one of our [community meetings][meetings]. If the created request doesn't include reference to a discussion and/or recordings of discussion in our community meetings, please note that it _may_ get closed
with a request to create an initial discussion first.
```

If you're unsure the proposed work requires an enhancement, file an issue
and ask.

> ## New Enhancement process

Reviewer 2: ```suggestion
## New Enhancement Process
```

Reviewer 2: @youngnick I think you may have missed pushing up the latest set of updates here.

Author: So I did. Fixed now.

> The new enhancement process is documented on the [GEP Overview][gep] page. Please

Reviewer 2: ```suggestion
The process for creating new enhancement proposals is documented on the [GEP Overview][gep] page. Please
```

see that page for all the details about how to log a new GEP, and the process
it will follow on its journey towards Completed status.

> A **documented** discussion of some form **must** exist prior to submitting a request for enhancement

Reviewer 3: Love the strong language here

Author: That one's from @shaneutt!

if that enhancement is non-trivial (which we will define as either: _implicates changes to the API specification_ OR _has some kind of end-user impact_). Please use our [Github Discussions][discussion]
forum as the initial place to start, and feel free to bring that discussion up for synchronous conversation in
one of our [community meetings][meetings]. If the created request doesn't include reference to a discussion and/or recordings of discussion in our community meetings, please note that it _may_ get closed

> with a request to create an initial discussion first.

Reviewer 2: Nit: Wrapping at 80 chars would be nice for consistency with the rest of the file.

Author: Turns out "accept suggestion" in the Github UI is not an ideal way to do this. Fixed now.
